cmd/dispatch-daemon/internal/daemon.go

```diff
@@ -0,0 +1,68 @@
+package internal
+
+import (
+	"context"
+	"subscription-api/config"
+	pb_ds "subscription-api/pkg/grpc/dispatch_service"
+	"time"
+
+	"google.golang.org/grpc"
+)
+
+type Scheduler interface {
+	AddTask(name string, spec TaskSpec, task func())
+	RemoveTask(taskId int)
+	Run()
+}
+
+type DispatchService interface {
+	SendDispatch(ctx context.Context, in *pb_ds.SendDispatchRequest, opts ...grpc.CallOption) (*pb_ds.SendDispatchResponse, error)
+	GetAllDispatches(ctx context.Context, in *pb_ds.GetAllDispatchesRequest, opts ...grpc.CallOption) (*pb_ds.GetAllDispatchesResponse, error)
+}
```

Comment:
This is very tricky interface =)
You use "abstraction" in face of interaface but now you depends on grpc variables, so this can be used only with grpc. I can tell like this is something like "concrete interface".

It should be something like 
```golang
 type DispatchService interface {
	SendDispatch(ctx context.Context, id int) error
	GetAllDispatches(ctx context.Context) (SomeStruct, error)
}
```

Reply:
Understood and updated interfaces due to comment